integration-tests/test-app/focus-input.ts

```diff
@@ -0,0 +1,15 @@
+import { registerHtml, useEffect, TramOneComponent } from '../../src/tram-one';
+
+const html = registerHtml();
+
+/**
+ * Element to test useEffect reference point
+ */
+const focusInput: TramOneComponent = () => {
+	useEffect((ref) => {
+		(ref as unknown as HTMLElement).focus();
```

Comment:
frustratingly, since `ref` is of type TramOneElement, there's no guarantee that it's an HTMLElement or SVGElement - so we have to do this casting to unknown. 

I'll look into ways we can maybe avoid this, potentially having generics in the registerDOM parts of the code.

Reply:
Fixed this by adding a ton of type definitions around the dom-wrappers logic.